core/domain/cron_services.py

```diff
@@ -30,38 +38,38 @@
 MODEL_CLASSES_TO_MARK_AS_DELETED = (user_models.UserQueryModel,)
 
 
-def delete_models_marked_as_deleted():
+def delete_models_marked_as_deleted() -> None:
     """Hard-delete all models that are marked as deleted (have deleted field set
     to True) and were last updated more than eight weeks ago.
     """
     date_now = datetime.datetime.utcnow()
     date_before_which_to_hard_delete = (
         date_now - feconf.PERIOD_TO_HARD_DELETE_MODELS_MARKED_AS_DELETED)
     for model_class in models.Registry.get_all_storage_model_classes():
-        deleted_models = model_class.query(
+        deleted_models: Sequence[base_models.BaseModel] = model_class.query(
             model_class.deleted == True  # pylint: disable=singleton-comparison
         ).fetch()
-        models_to_hard_delete = [
+        models_to_hard_delete: List[base_models.BaseModel] = [
             deleted_model for deleted_model in deleted_models
             if deleted_model.last_updated < date_before_which_to_hard_delete
         ]
         if issubclass(model_class, base_models.VersionedModel):
-            model_ids_to_hard_delete = [
+            model_ids_to_hard_delete: List[str] = [
                 model.id for model in models_to_hard_delete
             ]
-            model_class.delete_multi(
+            base_models.VersionedModel.delete_multi(
```

Comment:
@vojtechjelinek, initially it looks like the backend test failure is triggered by this change, and if we look into the failing test `cron_test.CronJobTests.test_run_cron_to_hard_delete_versioned_models_marked_as_deleted` on line 126, we can see that after calling the `CronModelsCleanupHandler` handler from `cron.py`, we are expecting `None` for the deletion of those models( marked as deleted ). So, I will try to find out why those models are not deleted. I am also open for your suggestion. Thanks,